keystore/pkcs12.go

```diff
@@ -0,0 +1,39 @@
+package keystore
+
+import (
+	"crypto/x509"
+	"fmt"
+
+	"github.com/bitrise-io/go-pkcs12"
+)
+
+type PKCS12KeystoreDecoder struct {
+}
+
+func (d PKCS12KeystoreDecoder) Decode(data []byte, password, alias, keyPassword string) (privateKey interface{}, certificate *x509.Certificate, err error) {
+	key, cert, err := pkcs12.DecodeKeystore(data, password, alias, keyPassword)
+	if err != nil {
+		switch err {
+		case pkcs12.IncorrectKeystorePasswordError:
+			return nil, nil, IncorrectKeystorePasswordError
+		case pkcs12.IncorrectAliasError:
+			return nil, nil, IncorrectAliasError
+		case pkcs12.IncorrectKeyPasswordError:
+			return nil, nil, IncorrectKeyPasswordError
+		default:
+			return nil, nil, err
+		}
+	}
```

Comment:
nit: This embedded switch somehow made this section hard to read for me. I think the error mapping is not that important when you are reading the flow of this function so I would propose to extract it into a separate method. Something like this:
```
func (d PKCS12KeystoreDecoder) Decode(data []byte, password, alias, keyPassword string) (privateKey interface{}, certificate *x509.Certificate, err error) {
	...
	if err != nil {
		return nil, nil, keystoreErrorFromPkcs12Error(err)
	}
	...
}

func keystoreErrorFromPkcs12Error(err error) error {
	switch err {
	case pkcs12.IncorrectKeystorePasswordError:
		return IncorrectKeystorePasswordError
	case pkcs12.IncorrectAliasError:
		return IncorrectAliasError
	case pkcs12.IncorrectKeyPasswordError:
		return IncorrectKeyPasswordError
	default:
		return err
	}
}
```

Reply:
Updated in https://github.com/bitrise-io/go-android/pull/55/commits/b06082080b5f6f3608d172055f5349b7abc99c01